ls1intum/Artemis
# [Bugfix] Various Bugs

### Checklist

### Motivation and Context
During the testing sessions last week we found a few bugs. This PR fixes #2503, fixes #2502, fixes #2493, fixes #2530, fixes #2496

### Description
#2503 => Add TA authority to route
#2502 => Fix route to student summary for instructor access
#2493 => Handle undefined values appropriately
#2496 => Limit text submissions to 30.000 characters
Remove QA button from exam assessment dashboard

### Steps for Testing
Instructors could not access the student's exam summary

    1. Go to an exam with student exams
    2. Click on 'Student Exams'
    3. Click on 'View' for a student exam
    4. Click on 'Summary'
    5. No error should be thrown, the exam summary should show

Tutors cannot access the exam management page

    1. Go to 'course management' as a tutor (no admin or instructor rights!)
    2. Click on 'Exams of a course'
    3. The exam management screen should show, no error should be thrown

Empty text field in short answer questions causes an incorrect behavior in the submission review

    1. Create a quiz with short answer questions and other exercises coming after the quiz
    2. Participate in the exam as a student, leave the text input field empty
    3. Wait for the quiz to end (reload if not automatically)
    4. The quiz should be shown correctly, all example submissions should load as expected

    1. If you created your quiz as part of an exam, you must first click on "evaluate quizzes" in the Student Exams tab, and set the release date of the results before you can view your score in the summary.

Limit text submissions to 30.000 characters

    1. Create a text exercise
    2. Participate in text exercise
    3. Try to submit a text with over 30k characters: https://www.blindtextgenerator.com/lorem-ipsum Observe error
    4. Submit a text with 30k characters
    5. Assess submission
    6. Complain about submission
    7. Evaluate complaint

Remove QA button from exam assessment dashboard

    1. Go to 'course management' as a tutor or instructor
    2. Click on 'Exams of a course'
    3. Click on Assessment Dashboard
    4. No QA button should appear in the header

### Test Coverage
No changes
